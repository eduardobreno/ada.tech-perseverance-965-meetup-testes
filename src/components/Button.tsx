import React from 'react';
import {TouchableOpacity, Text, StyleSheet} from 'react-native';

interface ButtonProps {
  testID: string;
  title: string;
  onPress: () => void;
}

const Button = ({title, onPress, testID = ''}: ButtonProps) => (
  <TouchableOpacity testID={testID} style={styles.button} onPress={onPress}>
    <Text>{title}</Text>
  </TouchableOpacity>
);

const styles = StyleSheet.create({
  button: {
    borderColor: '#000000',
    borderWidth: 1,
    paddingVertical: 10,
    paddingHorizontal: 25,
  },
});

export default Button;
