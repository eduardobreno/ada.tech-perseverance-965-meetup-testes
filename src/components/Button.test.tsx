import React from 'react';
import {render, screen, fireEvent} from '@testing-library/react-native';
import Button from './Button';

it('should press button', async () => {
  const add = jest.fn();
  render(<Button title="Adicionar" onPress={add} />);

  fireEvent.press(screen.getByText('Adicionar'));

  expect(add).toBeCalled();
  expect(screen.toJSON()).toMatchSnapshot();
});
