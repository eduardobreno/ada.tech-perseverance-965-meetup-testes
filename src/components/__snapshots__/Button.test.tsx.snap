// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should press button 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "borderColor": "#000000",
      "borderWidth": 1,
      "opacity": 1,
      "paddingHorizontal": 25,
      "paddingVertical": 10,
    }
  }
  testID=""
>
  <Text>
    Adicionar
  </Text>
</View>
`;
